File: php/math/sieve.php
<?php
// Print the primes below 100, sieved over an array of flags and read back with array_keys.

declare(strict_types=1);

const LIMIT = 100;

$isPrime = array_fill(0, LIMIT, true);
$isPrime[0] = $isPrime[1] = false;

for ($n = 2; $n * $n < LIMIT; $n++) {
    if (!$isPrime[$n]) {
        continue;
    }
    for ($multiple = $n * $n; $multiple < LIMIT; $multiple += $n) {
        $isPrime[$multiple] = false;
    }
}

echo implode(' ', array_keys($isPrime, true, true)), "\n";

File: r/math/sieve.R
# Print the primes below 100, sieved over a logical vector read back with which.
# Run: Rscript sieve.R

limit <- 100

is_prime <- rep(TRUE, limit)
is_prime[1] <- FALSE

for (n in 2:floor(sqrt(limit))) {
  if (is_prime[n]) {
    is_prime[seq(n * n, limit, by = n)] <- FALSE
  }
}

cat(paste(which(is_prime), collapse = " "), "\n", sep = "")
